### DISCUSSION TITLE
Route Aliases

### DISCUSSION THREAD

**Discussion by populated** [pallets/flask#5419] [Q&A]
I think it would be cool if there were built-in aliases for routes, for example:

```py
@route(
    "/information",
    aliases=["/info", "/faq"] # <- They all show the same thing as the main route `/information`.
)
```

It might be useless to some, but I don't. I think it would be useful to allow people to define multiple aliases for one route. It is possible as of now using this code I made:

```py
from typing import List, Callable, Union
from flask import Flask, wraps, redirect, jsonify

app = Flask(__name__)

def route(name: str, aliases: List[str] = [], **options) -> Callable:
    def decorator(func: Callable) -> Callable:
        @app.route(name, **options)
        @wraps(func)
        async def wrapper(*args, **kwargs) -> Union[redirect, jsonify]:
            return await func(*args, **kwargs)

        [app.route(alias, **options)(wrapper) for alias in aliases]

        return wrapper

    return decorator
```

**Comment by ThiefMaster** [comment]
Your aliases should probably redirect and not result in multiple URLs doing the same thing... it's generally considered bad to have duplicate content on the web w/o redirects to the canonical URL.
Also, don't abuse list comprehensions where a for loop would be appropriate.

In any case, this is something you can easily implement in your own app if you want it.

**Comment by populated** [comment]
> Your aliases should probably redirect and not result in multiple URLs doing the same thing... it's generally considered bad to have duplicate content on the web w/o redirects to the canonical URL. Also, don't abuse list comprehensions where a for loop would be appropriate.
> 
> In any case, this is something you can easily implement in your own app if you want it.

My code was something I created on the spot. I haven't taken a comprehensive look at it, as you mentioned, duplicate content on the web. I mean, would it be better if I made them all redirect back to the main route?

**Comment by tachyondecay** [comment]
You can already achieve your original ask by simply adding additional `URL rules to an endpoint:

```py
@app.route("/foo")
@app.route("/bar")
def hey():
    return "hello world"
````

Yes, it’s better to have a single, canonical URL for a page. It’s highly unusual to have a single, static page accessible from multiple URLs.

**Reply by populated** [reply]
So, this **new** code should achieve that:

```py
def route(name: str, aliases: List[str] = [], **options) -> Callable:
    def decorator(func: Callable) -> Callable:
        @app.route(name, **options)
        @wraps(func)
        async def wrapper(*args, **kwargs) -> Union[redirect, jsonify]:
            return await func(*args, **kwargs)

        for alias in aliases:
            app.add_url_rule(alias, endpoint=wrapper.__name__, view_func=wrapper, **options)

        return wrapper

    return decorator
```
